public/assets/ts/exam-bank.ts

```diff
@@ -0,0 +1,139 @@
+// @todo use a single source of the Exam type for frontend and backend
+type Exam = {
+  department: string;
```

Comment:
this might be overkill but maybe we should create enums for `department` and `type` since we know their values will always be in a specific set? like 
```
enum Department {
MATH = 'MATH',
ACTSC = 'ACTSC',
...
}
```
and then we could do `department: Department` instead of `department: string`?

Reply:
I'd love it if we always knew their values were in the specific set, but I don't think that we do!  We could build the enum based on current exams' departments, but then what happens if a future VPA adds an exam from an hidden-away arts faculty sub-department and breaks our type safety? 

With `type`, the issue's less that it's infeasible and more that it might not be that valuable. Every course has different naming conventions for their tests and exams - we could enforce that each exam falls into quiz/midterm/final and map the existing exams accordingly, but I'm not sure what the value of it would be.  Since students search by course name, they don't really need well-defined data structures for the type of exam as long as it's readable.

lmk if you think we should enforce the quiz types and I can work on it or push it to another ticket